crates/astria-conductor/src/executor/mod.rs

```diff
@@ -45,11 +52,83 @@ mod tests;
 pub(super) use client::Client;
 pub(crate) use state::State;
 
+#[derive(Clone, Debug)]
+pub(crate) struct StateNotInit;
+#[derive(Clone, Debug)]
+pub(crate) struct StateIsInit;
+
+/// A handle to the executor.
+///
+/// To be be useful, [`Handle<StateNotInit>::wait_for_init`] must be called in
+/// order to obtain a [`Handle<StateInit>`]. This is to ensure that the executor
+/// state was primed before using its other methods. See [`State`] for more
+/// information.
+#[derive(Debug, Clone)]
+pub(crate) struct Handle<TStateInit = StateNotInit> {
+    firm_blocks: mpsc::UnboundedSender<ReconstructedBlock>,
+    soft_blocks: mpsc::UnboundedSender<SequencerBlock>,
+    state: watch::Receiver<State>,
+    _state_init: TStateInit,
```

Comment:
is this instead of a `PhantomData`?

Reply:
Yes. But since we are only using zero sized unit types there is no advantage to using `PhantomData`, so this is simpler.